src/main/resources/alma/fix/subjects.fix

```diff
@@ -0,0 +1,397 @@
+#     <!-- 9 -->
+#     <call-macro name="schlagwortfolge" field="6898" />
+##     <!-- 10 -->
+##     <call-macro name="schlagwortfolge" field="6899" />
+##   </entity>
+# <macro name="schlagwortfolge">
+#    <entity name="" flushWith="record">
+#      <entity name="type[]" sameEntity="true" reset="true">
+#        <data name="" source="$[field]0">
+#          <constant value="ComplexSubject"/>
+#        </data>
+#      </entity>
+#      <concat name="label" delimiter=" | ">
+#        <data source="$[field]?.a" />
+#      </concat>
+#      <entity name="componentList[]" flushWith="record">
+#        <entity name="" sameEntity="true" reset="true" flushWith="$[field]?">
+#          <entity name="type[]" sameEntity="true" reset="true">
+#            <data source="$[field]?.D">
+#              <lookup in="rswk-indicator" />
+#            </data>
+#          </entity>
+#          <data name="label" source="$[field]?.a" />
+#          <data name="id" source="$[field]?.0">
+#            <regexp match="\(DE-588\)(.*)" format="http://d-nb.info/gnd/${1}"/>
+#          </data>
+#          <entity name="source" sameEntity="true" reset="true">
+#            <data name="id" source="$[field]?.0">
+#              <regexp match="\(DE-588\)(.*)" format="https://d-nb.info/gnd/7749153-1"/>
+#            </data>
+#            <data name="label" source="$[field]?.0">
+#              <regexp match="\(DE-588\)(.*)" format="Gemeinsame Normdatei (GND)"/>
+#            </data>
+#          </entity>
+#          <data name="gndIdentifier" source="$[field]?.0">
+#            <regexp match="\(DE-588\)(.*)" format="${1}"/>
+#          </data>
+#        </entity>
+#      </entity>
+#    </entity>
+#  </macro>
+#
+
+if exists("6890?")
+  set_array("subject[].$append.type[]","ComplexSubject")
+  set_array("subject[].$last.label")
+  set_array("subject[].$last.componentList[]")  
+  do list(path:"6890?", "var":"$i")
+    set_array("subject[].$last.componentList[].$append.type[]")
+    do list(path: "$i.D", "var": "$k")
+      copy_field("$k","subject[].$last.componentList[].$last.type[].$append")
+    end
+    copy_field("$i.a","subject[].$last.componentList[].$last.label")
+      do list(path:"$i.0", "var": "$j")
+      if any_match("$j","^\\(DE-588\\)(.*)$")
+        add_field("subject[].$last.componentList[].$last.source.label","Gemeinsame Normdatei (GND)")
+        add_field("subject[].$last.componentList[].$last.source.id","https://d-nb.info/gnd/7749153-1")
+        copy_field("$j", "subject[].$last.componentList[].$last.id")
+        replace_all("subject[].$last.componentList[].$last..id","^\\(DE-588\\)(.*)$","http://d-nb.info/gnd/$1")
+        copy_field("$j", "subject[].$last.componentList[].$last.gndIdentifier")
+        replace_all("subject[].$last.componentList[].$last..gndIdentifier","^\\(DE-588\\)(.*)$","$1") 
+      end
+    end
+  end
+end
+
+# TODO: Find out why this does not work?
+# lookup("subject[].*.componentList[].*.type[].*","rswk-indicator")
+
```

Comment:
@blackwinter do you have an idea why this breaks?

Reply:
```
Caused by: java.lang.IllegalArgumentException: Using ref, but can't find: 2 in: null
        at org.metafacture.metafix.FixPath.insertInto(FixPath.java:298)
        at org.metafacture.metafix.FixPath.insertInto(FixPath.java:265)
        at org.metafacture.metafix.FixPath.lambda$insertInto$20(FixPath.java:293)
        at org.metafacture.metafix.Value$TypeMatcher.match(Value.java:344)
        at org.metafacture.metafix.Value$TypeMatcher.ifArray(Value.java:317)
        at org.metafacture.metafix.FixPath.lambda$insertInto$22(FixPath.java:293)
```

Reply:
It runs with: `$ mvn clean install -DskipTests=true ; mvn -DgenerateTestData=true failsafe:integration-test -Dit.test=AlmaMarc21XmlToLobidJsonMetafixTest.java`

Reply:
I'm afraid I have no quick fix here, it's actually surprising how that current path update logic works at all, I'll keep trying.

Reply:
Thanks this seems to work now!